### ISSUE TITLE
Default parameter capability

### ISSUE THREAD

**Issue by emrecengel** [canton7/RestEase#70]
I believe one of the missing feature is adding default parameter to context, such as how header works in this library, which will be included on each request automatically. That would be a great feature, I understand there are ways to accomplish this, but including it as attribute would make it much easier to manage.

Thanks in advance.

**Comment by canton7** [comment]
What sort of parameters? Query? Path? Body?

**Comment by emrecengel** [comment]
Yeah, I would say [Query] and [Path] would be nice to have, if you would include [Body], I see no harm having it as well.

**Comment by canton7** [comment]
I don't think [Path] makes sense, since you can just write:

```csharp
interface IFoo
{
    [Path("foo/bar")]
    Task FooAsync()
}
```

instead of:

```csharp
[Path("barPlaceholder", "bar"]
interface IFoo
{
    [Path("foo/{barPlaceholder}")]
    Task FooAsync()
}
```

Do you actually have a use-case for [Query], or is this just for symmetry?

**Comment by emrecengel** [comment]
I wouldn't worry about [Path], but I think having [Query] is important. 

First of all the idea behind it to limit code repetition, let's say the api you are consuming and needs a version number or like session key in all calls and let this key is inner changeable based on consumer at that point.
 

```cs
        [Get("Contact")]
        Task<Response<List<Contact>>> Contacts([Query("uniqueValue")] string uniqueValue);
```
think about a case where you have to include this for 100+ calls.

if this would be like how `[Header]` works,  no need to include it all of the methods.

```cs
        [Query("uniqueValue")]
        string UniqueValue { get; set; }

        [Get("Contact")]
        Task<Response<List<Contact>>> Contacts();
```

**Comment by canton7** [comment]
Oh, you're talking about query _properties_, rather than applying [Query] to the entire interface? When you said "but including it as attribute would make it much easier to manage", I assumed you were referring to an attribute on the entire interface.

**Comment by emrecengel** [comment]
I am sorry for the confusion, yeah I meant having it as property that wrapped with Attribute like [Header].

**Comment by canton7** [comment]
Aah right, I see. I was also confused by:

> Yeah, I would say [Query] and [Path] would be nice to have,

As we already have path properties.
